HierarchicalDataConverter/Catalyst.HierarchicalDataConverter/RestApiDestinationSystem.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace DataConverter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Catalyst.DataProcessing.Engine.PluginInterfaces;
+    using Catalyst.DataProcessing.Shared.Models.DataProcessing;
+    using Catalyst.DataProcessing.Shared.Models.Metadata;
+    using Catalyst.DataProcessing.Shared.Utilities.Client;
+
+    /// <summary>
+    /// Databus currently handles the promotion of data to its final destination (Rest API), therefore this is a no op implementation
+    /// </summary>
+    public class RestApiDestinationSystem : IDestinationSystem
+    {
+        private readonly IMetadataServiceClient metadataServiceClient;
+
+        public RestApiDestinationSystem(IMetadataServiceClient metadataServiceClient)
+        {
+            this.metadataServiceClient = metadataServiceClient ?? throw new ArgumentException("metadataServiceClient cannot be null.");
+        }
+
+        public async Task AddEntityOptimizationsAsync(EntityExecution entityExecution, Entity entity, DataMart dataMart, CancellationToken cancellationToken)
+        {
+            // no op
+            return;
+        }
+
+        public bool CanHandle(Entity entity)
+        {
+            Binding[] bindings = this.metadataServiceClient.GetBindingsForEntityAsync(entity.Id).Result;
+            return bindings.Any(b => b.BindingType == HierarchicalDataTransformer.NestedBindingTypeName);
```

Comment:
Null check?

Reply:
updated

Reply:
Thanks. Was also thinking the Binding[] bindings coule be null. If so, Any will throw a ArgumentNull exception. In this case, seems like return false?

Reply:
I don't think the metadataServiceClient can return null, but I added the extra check anyway - can't hurt